Snail/Observable.swift

```diff
@@ -216,6 +216,28 @@ public class Observable<T>: ObservableType {
         return observable
     }
 
+    public func take(first count: UInt) -> Observable<T> {
+        let observable = Observable<T>()
+        var taken = 0
+
+        subscribe(onNext: {
+            if taken < count {
```

Comment:
should we tighten up the logic with if/else or guard?

Reply:
```
guard taken < count else {
    observable.on(.done)
}
observable.on(.next($0))
taken += 1
```
?

Reply:
What would that look like? If else here wouldn't work because we want `.done` to be called even if we just sent the last event.

Reply:
i didnt see how guard would work, but this looks nice @jacksoncheek

Reply:
actually this wont work ): since we update `taken` after the guard, the `.done` wont be sent even though we just took the last event.

Reply:
think what i have is the simplest even if I don't like the two ifs.